src/coreclr/jit/vartype.h

```diff
@@ -118,13 +118,15 @@ inline bool varTypeIsSigned(T vt)
     return varTypeIsIntegralOrI(vt) && !varTypeIsUnsigned(vt);
 }
 
-// If "vt" is an unsigned integral type, returns the corresponding signed integral type, otherwise
-// return "vt".
-inline var_types varTypeUnsignedToSigned(var_types vt)
+// If "vt" represents an unsigned integral type, returns the corresponding signed integral type,
+// otherwise returns the original type.
+template <class T>
+inline var_types varTypeToSigned(T vt)
```

Comment:
just bunch of thoughts, you can ignore them if you don't agree:
There is only one call to this function:
https://github.com/dotnet/runtime/blob/49eef91d24e282d2548827a601a9caa65882e499/src/coreclr/jit/compiler.h#L5187-L5188
and the semantic "otherwise returns the original type." is kind of confusing for me, I think I would prefer an `assert(varTypeIsUnsigned(vt))` in the function and `if (varTypeIsUnsigned(vt))` in the caller.
and maybe move this function to the only user?

Reply:
> the semantic "otherwise returns the original type." is kind of confusing for me

I agree. I can make it `assert`, but that will require changing `varTypeToUnsigned` too, for consistency, and that'll make the review significantly harder, so I suppose I would prefer to do it as a separate change. What would your preference be?

> maybe move this function to the only user?

Well, there are now 3 users, and I personally used it some more in some of the (now dead) changes to `fgMorphCast`, so I think it has its value. More broadly, I think it is useful to have these helpers, even if they are little used.